### PR TITLE
[Feature] Add LightCNN for DICGAN

### PR DESCRIPTION
**Pull request by Yshuo-Li**

### CODE REVIEW

File: mmedit/models/components/discriminators/light_cnn.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            self.filter = nn.Conv2d(
+                in_channels,
+                2 * out_channels,
+                kernel_size=kernel_size,
+                stride=stride,
+                padding=padding)
+        elif filter_type == 'linear':
+            self.filter = nn.Linear(in_channels, 2 * out_channels)
+        else:
+            raise ValueError("'filter_type' should be 'conv2d' or 'linear',"
```
Comment (ckkelvinchan) [inline mmedit/models/components/discriminators/light_cnn.py:46]: Add a space after `'linear',` so that there will be a space between `,` and `but`.

File: mmedit/models/components/discriminators/light_cnn.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+        Args:
+            x (Tensor): Input tensor.
+
+        Returns:
+            Tensor: Forward results.
+        """
+
+        x = self.filter(x)
+        out = torch.split(x, self.out_channels, 1)
```
Comment (ckkelvinchan) [inline mmedit/models/components/discriminators/light_cnn.py:60]: `out = torch.chunk(x, 2, dim=1)`. Then we do not need `self.out_channels` .

File: mmedit/models/components/discriminators/light_cnn.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            nn.MaxPool2d(kernel_size=2, stride=2, ceil_mode=True),
+            MaxFeature(96, 96, 1, 1, 0),
+            MaxFeature(96, 192, 3, 1, 1),
+            nn.MaxPool2d(kernel_size=2, stride=2, ceil_mode=True),
+            MaxFeature(192, 192, 1, 1, 0),
+            MaxFeature(192, 128, 3, 1, 1),
+            MaxFeature(128, 128, 1, 1, 0),
+            MaxFeature(128, 128, 3, 1, 1),
+            nn.MaxPool2d(kernel_size=2, stride=2, ceil_mode=True),
+        )
```
Comment (ckkelvinchan) [inline mmedit/models/components/discriminators/light_cnn.py:91]: Can add `nn.Flatten()`, and then combine `self.features` and `self.classifier`.

Comment (ckkelvinchan) [reply mmedit/models/components/discriminators/light_cnn.py:91]: I saw `self.features` is used in FeatureLoss. Ignore this.

**Review by ckkelvinchan — APPROVED** [review]
